### PR TITLE
Remove old manpages from contrib/debian in favour of doc/man

### LINKED ISSUE #8711: Duplicate man pages
#8608 added man pages to doc/man, however we already have a bunch of man pages in contrib/debian/manpages. These should be unified to only exist in one place.

### PR DESCRIPTION
**Pull request by fanquake**
Removes the old manpages from contrib/debian/manpages, in favour of the newer ones in doc/man.
Also adds a note about updating the manpages to release-process.md

Pinging @TheBlueMatt, would you be able to look this look this over and point out any fixes.

Fixes #8711

### CODE REVIEW

**Comment by MarcoFalke** [comment]
utACK 365c0920e89635f0605a1e604f489b363538a987

File: contrib/debian/bitcoind.manpages [hunk 1]
```diff
@@ -1,3 +1,4 @@
-debian/manpages/bitcoind.1
-debian/manpages/bitcoin.conf.5
-debian/manpages/bitcoin-cli.1
+doc/man/bitcoind.1
+doc/man/bitcoin-cli.1
+doc/man/bitcoin-qt.1
```
Comment (TheBlueMatt) [inline contrib/debian/bitcoind.manpages]: This should be in a contrib/debian/bitcoin-qt.manpages (and similarly for bitcoin-tx)

File: contrib/debian/rules [hunk 2]
```diff
@@ -6,7 +6,7 @@
 #	$(if $(filter nocheck,$(DEB_BUILD_OPTIONS)),,src/test_bitcoin)
 
 DEB_INSTALL_EXAMPLES_bitcoind += debian/examples/*
-DEB_INSTALL_MANPAGES_bitcoind += debian/manpages/*
+DEB_INSTALL_MANPAGES_bitcoind += doc/man/*
```
Comment (TheBlueMatt) [inline contrib/debian/rules]: Actually, these two lines should just be wholesale removed, I believe.

**Comment by laanwj** [comment]
utACK after fixing @thebluematt's nits

File: contrib/debian/rules (hunk 2 above)
Comment (laanwj) [inline contrib/debian/rules]: I think so too - our installer installs the man pages, there's no need for debian to do anything special.

Comment (TheBlueMatt) [inline contrib/debian/rules]: No, mostly this is a flag for a build method that we dont use - so these variables are just entirely ignored :).

**Comment by laanwj** [comment]
@fanquake I've added a commit that fixes @TheBlueMatt 's comments. If this change is okay we should squash and merge this.

**Comment by fanquake** [comment]
@laanwj Thanks for fixing this up, I haven't had access to push the changes.

**Pull request merged**